src/HtmlGenerator/Pass1-Generation/DocumentGenerator.Links.cs

```diff
@@ -50,7 +51,9 @@ private HtmlElementInfo GenerateLinks(Classification.Range range, bool isLargeFi
             }
 
             // now that we've passed the empty array allocation check, disable all further new keywords
-            if (range.ClassificationType == Constants.ClassificationKeyword && text == "new")
+            // except for target-typed 'new' expressions
+            if (range.ClassificationType == Constants.ClassificationKeyword && text == "new" &&
+                !(token.Parent is ImplicitObjectCreationExpressionSyntax))

```

Comment:
could have prob used `is not` but no biggie

Reply:
I couldn't because:

```
Error	CS8370	Feature 'not pattern' is not available in C# 7.3. Please use language version 9.0 or greater.	
```

And I didn't want to mess with updating the LangVersion. I can update it in a separate change, if you think that would be acceptable.

Reply:
I'll switch the langversion to latest, not sure why it's pinned

Reply:
Ah, turns out it's pinned by default, weird. Fixed now, but don't worry about fixing the is not, I can do it later next time. Don't want to cause extra work for you.

Reply:
Is there any reason to keep using net472? What would you think about upgrading to net6.0?

I messed around for a bit updating to 6.0 here: https://github.com/eerhardt/SourceBrowser/pull/new/UpdateTo6. It isn't ready for a PR yet. I was doing it mostly to learn and feel the customer experience myself.

The current warnings I have:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	NU1701	Package 'Microsoft.VisualStudio.Language.Intellisense 16.10.230' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework 'net6.0'. This package may not be fully compatible with your project.	HtmlGenerator.Tests	C:\git\SourceBrowser\src\HtmlGenerator.Tests\HtmlGenerator.Tests.csproj	1	
Warning	NU1701	Package 'Microsoft.VisualStudio.Language.Intellisense 16.10.230' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework 'net6.0'. This package may not be fully compatible with your project.	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\HtmlGenerator.csproj	1	
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\DocumentGenerator.HighlightReferences.cs	8	Active
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\DocumentGenerator.Links.cs	361	Active
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\DocumentGenerator.References.cs	257	Active
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\MetadataAsSource.cs	93	Active
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\ProjectGenerator.cs	38	Active
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\ProjectGenerator.cs	39	Active
Warning	RS1024	Compare symbols correctly	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\Pass1-Generation\DocumentGenerator.cs	61	Active
```

The first 2 can be ignored - we are only using 2 enums from `Microsoft.VisualStudio.Language.Intellisense`. I was half thinking of just copying the enums into this code and removing the reference.

The `Compare symbols correctly` warnings sound like they should actually be fixed, since they are coming from a Roslyn analyzer. I haven't investigated them yet though.

Also, I am sometimes getting:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0006	Metadata file 'C:\git\SourceBrowser\src\MEF\bin\Debug\net6.0\ref\MEF.dll' could not be found	HtmlGenerator	C:\git\SourceBrowser\src\HtmlGenerator\CSC	1	Active
```

When building in VS. It doesn't happen consistently, and I haven't had a chance to look into it yet.

Reply:
I guess we publish a NuGet package `SourceBrowser` which contains the tool and many people run it in their CI which may or may not have net6. Also I really don't know what environments people are running this in, but I'm assuming there are lots of internal deployments of SourceBrowser and it'd be unreasonable to force them all to switch to net6.

One amazing thing about net472 is that every windows machine has it. I'm not sure how to achieve the same with net6.

We should still fix the compare symbols warnings I'm guessing.

Reply:
We could bundle the runtime of course, but that'd significantly increase the size of the package. At this point I'd rather not break anything if it isn't broken.